### DISCUSSION TITLE
Usage of cl.Development() and transform

### DISCUSSION THREAD

**Discussion by cooneycw** [casact/chainladder-python#541] [Q&A] [answered]
I am using the following code to create ldf's via the cl.Development() method:

model = cl.Development(average='simple').fit(triangle_df)

My understanding is that i can then use:

ult = model.transform(triangle_df)

to apply the ldf's and extend the triangle to ultimate via the factors.  This step executes without any message.   Next, when i attempt:

ult.full_triangle_

however, i am experiencing the following error:

AttributeError: 'Triangle' object has no attribute 'full_triangle_'

ult does show a triangle of data with NaN values equivalent to triangle_df.  Can someone pls help me understand what i am doing wrong?  I have also tried using:

 cl.Development(average = 'simple').fit_transform(triangle_df)

With the same outcome.  Many thanks.

**Comment by jbogaardt** [comment, marked as answer]
Development only produces development patterns, to get ultimates and a full triangle, you'd need to additionally fit one of the IBNR models (Chainladder, BornhuetterFerguson, etc). 

```python
dev = cl.Development(average='simple').fit_transform(triangle_df)
model = cl.Chainladder().fit(dev)
```
or more compactly:
```python
cl.Pipeline(
    [('dev', cl.Development(average='simple')),
     ('model', cl.Chainladder())]
).fit(triangle_df)
```

**Reply by cooneycw** [reply]
Thank you.  And were i to make adjustments to my ldf's via drop, etc., these would then flow through into the Chainladder step?

**Reply by jbogaardt** [reply]
Correct. Feel free to check this by accessing the `ldf_` property on the chainladder step to verify.
```python
pipe = cl.Pipeline(
    [('dev', cl.Development(average='simple')),
     ('model', cl.Chainladder())]
).fit(triangle_df)
print(pipe.named_steps.model.ldf_)
print(pipe.named_steps.dev.ldf_)
```
